library/src/commonTest/kotlin/com/gu/recipe/ScaleRecipeTest.kt

```diff
@@ -1,53 +1,49 @@
 import com.gu.recipe.scaleRecipe
 import kotlin.test.Test
 import kotlin.test.assertEquals
 
 class ScaleRecipeTest {
-
     @Test
-    fun `parse template`() {
-        val template =
-            Template("Bake at {\"temperatureC\": 180, \"temperatureFanC\": 160} for {\"min\": 30, \"max\": 40, \"unit\": \"minutes\"}.")
-        val parsed = parseTemplate(template)
-
-        assertEquals(
-            listOf(
-                TemplateElement.TemplateConst("Bake at "),
-                TemplateElement.OvenTemperaturePlaceholder(
-                    temperatureC = 180,
-                    temperatureFanC = 160
-                ),
-                TemplateElement.TemplateConst(" for "),
-                TemplateElement.QuantityPlaceholder(min = 30f, max = 40f, unit = "minutes"),
-                TemplateElement.TemplateConst(".")
+    fun `scale a recipe`() {
+        val recipeTemplate = SeverSideRecipe(
+            id = "test-recipe",
+            ingredientsTemplate = listOf(
+                IngredientsTemplateElement(
+                    ingredientsList = listOf(
+                        IngredientsTemplateIngredientsList(
+                            template = """{"min": 100, "max": 120, "unit": "g", "scale": true} of flour"""
+                        )
+                    )
+                )
             ),
-            parsed.elements
+            instructionsTemplate = listOf(
+                InstructionsTemplateElement(
+                    descriptionTemplate = """pre-warm the oven to {"temperatureC": 180, "temperatureFanC": 160}"""
+                )
+            )
         )
-    }
-
-    @Test
-    fun `scale a recipe`() {
-        val recipeTemplate = RecipeTemplate(
+        val expectedRecipe = ClientSideRecipe(
             id = "test-recipe",
-            title = "Test Recipe",
-            ingredients = listOf(Template("""{"min": 100, "max": 120, "unit": "g", "scale": true} of flour""")),
-            instructions = listOf(Template("""pre-warm the oven to {"temperatureC": 180, "temperatureFanC": 160}""")),
+            ingredients = listOf(
+                IngredientElement(
+                    ingredientsList = listOf(
+                        IngredientsListIngredientsList(
+                            text = "200-240 g of flour"
+                        )
+                    )
+                )
+            ),
+            instructions = listOf(InstructionElement(description = "pre-warm the oven to 180°C (160°C fan)"))
         )
-        val scaledRecipe = scaleRecipe(recipeTemplate, 2.0f)
+        val scaledRecipe = scaleRecipe(recipeTemplate, 2.0f, unit = IngredientUnit.Metric)
         assertEquals(
-            Recipe(
-                id = "test-recipe",
-                title = "Test Recipe",
-                ingredients = listOf("200-240 g of flour"),
-                instructions = listOf("pre-warm the oven to 180°C (160°C fan)")
-            ),
+            expectedRecipe,
```

Comment:
Can we have more different sets of test to check more cases or is this one should be okay?